### ISSUE TITLE
Bug in `christoffels` for matrices manifolds

### ISSUE THREAD

**Issue by luisfpereira** [geomstats/geomstats#1607]
### Describe the bug

`christoffels` do not work properly with matrices manifolds. The reason is the shape of `inner_product_derivative_matrix`, which is 4-d for matrices, whereas only 3d if base point is vector.

### Steps/Code to Reproduce

```python
from geomstats.geometry.matrices import Matrices

n = 3
space = Matrices(n, n)
metric = matrices.metric

point = space.random_point()
metric.christoffels(point)
```

### Expected Behaviour

No error is thrown.

### Actual Behaviour

Error is raised when computing first christoffels einsum due to wrong dimensions.

### Your environment

```shell
Not relevant.
```

P.S. I'm assuming christoffels are defined for this kind of metrics. If not, then the code structure need to be rethought, because it is very misleading.

**Comment by Jules-Deschamps** [comment]
Oh I see now what you meant with point as a matrix, indeed this is a problem. Could we not see matrices as a single vector of n*n entries? I feel like having matrices as points would break most of the functions no?
I'm not very familiar with git so do you expect me to try and fix this issue or did you just want me to understand what you said in the curvature pr?

**Comment by lpereira95** [comment]
We can consider matrices as $n\times n$ (or $n\times m) vectors. But is it mathematical correct? (This is my main question. In terms of codiing is relatively straightforward)

I open the issue to ensure we do not forget to correct this and to raise awareness that most of the implementations in #1606 will fail for this case, so we need to update the code there.

We can correct this one in a separate PR, but it probably needs to get merged first then #1606, because you will need it there.

**Comment by ninamiolane** [comment]
I think we should consider the matrices as dim-vectors, where dim is the dimension of the matrix manifold in question. The Riemannian metric matrix is then of shape dim x dim, and the rest follows.

For example, this is done in bi-invariant metric which is a way of defining a Riemannian metric of some matrix Lie groups. We can see that the metric_matrix is of shape dim x dim:
https://github.com/geomstats/geomstats/blob/d1aa07d97cd5aa6b18373eb91d64969700c32463/geomstats/geometry/invariant_metric.py#L55

--> Do you also have an error for this matrix manifolds?

In other words, we have to define a coordinate system on the matrix manifold, so that the tangent vectors (also of point type: matrix) can be expressed in this coordinate system: they become of point type: vector.

**Comment by ninamiolane** [comment]
Also, for matrix manifolds that are vector spaces: e.g. Matrices, SymmetricMatrices, SkewSymmetricMatrices, the christoffels are 0.
